frontend/BoardGUI.py

```diff
@@ -31,12 +33,22 @@ def draw_piece(self, piece, position):
         else:
             self.game.screen.blit(self.white_piece_image, position)
 
-    def getCellClicked(self):
+    def get_cell_clicked(self):
         current_cell = vec2(pg.mouse.get_pos()) // self.cell_size
         col, row = map(int, current_cell)
         left_click = pg.mouse.get_pressed()[0]
         if left_click:
-            print(self.board.board[row][col].state)
+            # print(self.backend_game.get_board(row,col), self.count)
+            self.count += 1
+
+            if not self.backend_game.placement_complete():
+                x = self.backend_game.piece_placement(row, col)
+                # print(x, "\n")
+                if x == 1:
+                    self.draw_piece(self.backend_game.get_turn(),current_cell * self.cell_size)
+                    self.backend_game.end_turn()
```

Comment:
We should break out these 2 lines of the function as it imposes on separation of concerns. These line could be in a condition for if `get_cell_clicked` returns a cell. It coud be at this place in the code https://github.com/Advanced-Software-Engineering-UMKC/Nine-Mens-Morris/blob/main/frontend/GameGUI.py#L29-L33. There may be a better way to do it as well.

Reply:
Hmm, yeah, we could just make another function to draw pieces and swap turns that can be called in the GameGUI when a valid cell is returned, like how the board is drawn.

Reply:
That might make programming this section more of a pain later, though -- piece placement, movement, flying, and removing pieces are all determined by clicking. I guess we could add a get_phase function in BoardGUI and call it later to determine exactly what should be happening with the returned cell.

Reply:
Adding a `get_phase` is a good idea.